Common/ExecutionModel/vtkThreadedImageAlgorithm.cxx

```diff
@@ -24,13 +23,104 @@
 #include "vtkObjectFactory.h"
 #include "vtkPointData.h"
 #include "vtkStreamingDemandDrivenPipeline.h"
+#include "vtkExtentTranslator.h"
+
+
+#define USE_SMP_DEFAULT_2D_BLOCK
+
+// the default 2d block is used to compare with tbb's own 2d splitting option
+#ifdef USE_SMP_DEFAULT_2D_BLOCK
+
+#include <tbb/parallel_for.h>
+#include <tbb/blocked_range2d.h>
+
+#else
+
+#include "vtkSMPTools.h"   
+
+#endif
+
+struct vtkImageThreadStruct
+{
+  vtkThreadedImageAlgorithm *Filter;
+  vtkInformation *Request;
+  vtkInformationVector **InputsInfo;
+  vtkInformationVector *OutputsInfo;
+  vtkImageData   ***Inputs;
+  vtkImageData   **Outputs;
+};
+
+#ifdef USE_SMP_DEFAULT_2D_BLOCK
+
+class vtkApplyTransformTBB
+{
+  vtkImageThreadStruct* Str;
+public:
+
+  vtkApplyTransformTBB(vtkImageThreadStruct* threadStruct)
```

Comment:
Hey @dgobbi  regarding the style of these two functions, according to the style guide rule 5

Only one public class per header file. Internal helper classes may be forward declared in header files, but can then only be defined in implementation files, ie using the PIMPL idiom.

The definition should be in the heads, however I am basing this style off of vtkSMPContourFilterManyPieces.  Which is the correct usage in this case?